CIP-0013/README.md

```diff
@@ -143,13 +162,21 @@ For background, see
   - [Apple Developer Docs > Defining a custom URL scheme for your app](https://developer.apple.com/documentation/xcode/defining-a-custom-url-scheme-for-your-app)
   - [React Native > Linking](https://reactnative.dev/docs/linking.html)
 
-### How URI delegation portfolio links supplement use of JSON files for the same purpose?
+### Rationale for payment links
+
+#### Why confine payment links to address and amount like BIP-21?
+
+BIP-21 is limited to only features Bitcoin supports. A similar feature for Ethereum would, for example, also support gas as an extra parameter. BIP-21 is easily extensible but we have to take precaution to avoid different wallets having different implementations of these features as they become available on Cardano. To get an idea of some extra features that could be added, consider this (still under discussion) proposal for Ethereum: [EIP-681](https://eips.ethereum.org/EIPS/eip-681)
+
+### Rationale for stake pool links
+
+#### How do URI delegation portfolio links supplement use of JSON files for the same purpose?
 
 URIs facilitate the "social element" of delegated staking and pool promotion through a socially familiar, easily accessible, and less centralised convention for sharing stake pool references and potential delegation portfolios without having to construct or host a JSON file.
 
 The processing of a JSON file delivered by a web server will depend highly on a user's platform and might not even be seen by the wallet application at all.  With a properly associated `web+cardano:` protocol, developers and users have another option available in case JSON files are not delivered properly to the wallet application.
```

Comment:
Could we better and more cleanly handle the varied methods of delegation via a link that would "point" to a well-defined JSON/CBOR schema metadata file that can be fetched? This would greatly simplify the URIs themselves and the associated HTML where they are embedded in addition to improving troubleshooting and diagnostics.

Reply:
In the months it took the stake poll links to be put through as an addition to this CIP, the counter of a JSON file was the main objection.  Rather than look at the stake pool approach on its own merit, most CIP editors at the time and IOG developers who jumped in said that JSON would be "better" and for a while the discussion just ended there: let alone being open to a cooperative approach.  `//stake` was eventually merged when I managed to show that a rigorously defined URI syntax had fewer practical limitations than the JSON proponents had all assumed.

That is at least an indicator that a hybrid URI-to-JSON stake portfolio protocol would be unpopular.  In fact, over 2 years after those discussions, CIP-0017 remains unimplemented as far as I know

The motivation for having the one-line, simple syntax available is for the sake of web designers, SPOs, and marketers & is already in the currently defined CIP.  Other URLs (like encapsulated JSON file pathnames) or resource locators might be helpful... though still no commercial support for this kind of robust delegation that I know of.  If & when the time comes, we can suggest to those implementors that they create a new authority (e.g. `//stakelink`) with a CIP for it.

Reply:
... some dev will come along someday wanting to put urlencoded CBOR in a URL: so rather than create a precedent to support this in a constantly modified (or deprecated) CIP-0013 we can simply tell them, "Create a new `authority` keyword for it & define your support for it there" ... whether it relates to payments, delegation and/or anything else 🤓

Reply:
I suppose my primary objection is to the use of potentially unicode pool tickers as the "keys" in the URL arguments, I feel like this creates a cumbersome barrier to implementation for the wallets. However, I also understand this was introduced most likely in response to URL max-length limitations that could quickly be exceeded by 3+ pool hex IDs.

The parser that I've written to handle logical parsing of CIP-13 URIs does manage to handle it but has not been rigorously tested against things like URL-encoded UTF-8 tickers so will need some additional work there for matching.

https://github.com/Crypto2099/POO/blob/main/packages/lib/cardano-poo-ts/src/CIP13Parser.ts

Reply:
The problem with _not_ using the stake pools as keys is that the syntax blows up, and then fails for multi pool links, when using them as values.

After months of debate, those against using the stake pools as keys couldn't come up with an alternative syntax which was concise _and_ unambiguous:

- https://github.com/cardano-foundation/CIPs/pull/65#issuecomment-783159961
- https://github.com/cardano-foundation/CIPs/pull/61#issuecomment-782141931
- https://github.com/cardano-foundation/CIPs/pull/61#issuecomment-782664031

I don't know if your alternative syntax with key-value pairs would have this same problem, but the main problem with the last suggestion (having a `poolticker` keyword with the hex pool ID as the value`) would prevent multi-pool delegation links, since URI query variables can't have more than one value:

- https://github.com/cardano-foundation/CIPs/pull/61#issuecomment-782732129

So as difficult as it might be for parsers and wallets to implement this (as already pointed out by @SEEEBBBA in the earlier threads, without giving any specifics), I don't see a way around it unless someone wants to spell it out an alternative syntax & attempt (again) another deprecation of the CIP-13 stake pool link protocol as it's been defined for the last 2 years... keeping in mind that the main problem would be the support of _multiple pool values_ rather than the length of the URI string.